gateway/analytics.go

```diff
@@ -128,10 +128,17 @@ func (r *RedisAnalyticsHandler) recordWorker() {
 			suffix := rand.Intn(10)
 			analyticKey = fmt.Sprintf("%v_%v", analyticKey, suffix)
 		}
+
+		timeout := time.NewTimer(recordsBufferFlushInterval)
+
 		readyToSend := false
 		select {
 
 		case record, ok := <-r.recordsChan:
+			if !timeout.Stop() {
+				<-timeout.C
```

Comment:
Can you explain changes in this file? 
I'm not sure what it fix here.

Reply:
This resolves a resource leak with time.After usage, pointed out here: https://github.com/TykTechnologies/tyk/pull/4035#discussion_r868934148, explained on godoc: https://pkg.go.dev/time#NewTimer -- it's the only adressable production code time.After usage, so it made sense to fix it immediately.